scripts/lustre

```diff
@@ -89,6 +90,32 @@ lustre_validate_all() {
     return $OCF_SUCCESS
 }
 
+# If $hook_name file exists, run it.  Used to provide {pre,post}-{start,stop}
+# hooks.  The executables used for hooks must adhere to the OCF Resource Agent
+# Developer's guide with regards to exit codes and general behavior.
+lustre_hook() {
+    hook_name=$1
+
+    if [ "$hook_name" != "lustre_pre_start" ]   && \
+        [ "$hook_name" != "lustre_post_start" ] && \
+        [ "$hook_name" != "lustre_pre_stop" ]   && \
+        [ "$hook_name" != "lustre_post_stop" ]; then
+        exit $OCF_ERR_GENERIC
```

Comment:
This seems odd to me.  It is basically an assert(), because this can never actually happen at run time.

Actually, I'm not terribly a fan of this lustre_hook() function and the hard coded scripts.  I think each possible script should have its own $OCF_RESKEY_ variable.  If the variable is specified, try to run the script that is specified.  Also, when the script is called it should really be passed at least $OCF_RESKEY_dataset and $OCF_RESKEY_mountpoint.  Keep in mind that these scripts are going to be called for each individual lustre filesystem target, not just once per node.

I also agree with Olaf's comment in the ticket that we should not have all four script points right away.  We should only call the scripts were absolutely necessary.